primitives/src/block.rs

```diff
@@ -365,11 +365,89 @@ mod tests {
+        assert_eq!(version.to_consensus(), 1234567890);
+    }
+
+    // Test the creation of a dummy block
+    #[cfg(feature = "alloc")]
+    #[test]
+    fn block_creation() {
+        use crate::locktime::absolute;
+        use crate::merkle_tree::TxMerkleNode;
+        use crate::{transaction, Amount, Sequence, Witness};
+        use crate::script::ScriptBuf;
+
+        let create_dummy_tx = |txid, version, lock_time, sequence| Transaction {
+            version,
+            lock_time: absolute::LockTime::from_consensus(lock_time),
+            input: vec![transaction::TxIn {
+                previous_output: transaction::OutPoint { txid, vout: 0 },
+                script_sig: ScriptBuf::new(),
+                sequence: Sequence(sequence),
+                witness: Witness::new(),
+            }],
+            output: vec![transaction::TxOut {
+                value: Amount::ONE_SAT,
+                script_pubkey: ScriptBuf::new(),
+            }],
+        };
+
+        let txid1 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[1]).to_byte_array());
+        let txid2 =
+            transaction::Txid::from_byte_array(hashes::sha256::Hash::hash(&[2]).to_byte_array());
+        let tx1 = create_dummy_tx(txid1, transaction::Version::ONE, 2, 1);
+        let tx2 = create_dummy_tx(txid2, transaction::Version::TWO, 3, 2);
+        let transactions = vec![tx1.clone(), tx2.clone()];
+
+        let header = Header {
+            version: Version::ONE,
+            prev_blockhash: BlockHash::from_byte_array([0x99; 32]),
+            merkle_root: TxMerkleNode::from_byte_array([0x77; 32]),
+            time: 2,
+            bits: CompactTarget::from_consensus(3),
+            nonce: 4,
+        };
+        let witness_root = Some(WitnessMerkleNode::from_byte_array([0x88; 32]));
+        let block = Block::new_unchecked(header, transactions.clone()).assume_checked(witness_root);
+
+        assert_eq!(block.transactions(), transactions);
+        assert_eq!(block.cached_witness_root(), witness_root);
+        assert_eq!(block.header(), &header);
+        assert_eq!(block.block_hash(), header.block_hash());
+    }
+
+    // Check header size
+    #[test]
+    fn header_size() {
+        let header = Header {
+            version: Version::ONE,
+            prev_blockhash: BlockHash::from_byte_array([0x99; 32]),
+            merkle_root: TxMerkleNode::from_byte_array([0x77; 32]),
+            time: 2,
+            bits: CompactTarget::from_consensus(3),
+            nonce: 4,
+        };
+        let header_size = header.version.to_consensus().to_le_bytes().len()
+            + header.prev_blockhash.as_byte_array().len()
+            + header.merkle_root.as_byte_array().len()
+            + header.time.to_le_bytes().len()
+            + header.bits.to_consensus().to_le_bytes().len()
+            + header.nonce.to_le_bytes().len();
+
+        assert_eq!(header_size, Header::SIZE);
```

Comment:
It took me a minute  to work out that this test is checking that consensus serialization of a header matches our const `SIZE` value. Did you look at consensus code to work this out or did you copy the comment (the sum to 80 bit)? Either way, this is the sort of time you can throw a quick comment saying why the test is written how it is. Saves the next guy reading it wondering.

Reply:
Comments added as suggested

Reply:
"Block headers are 80 bytes" is common knowledge in the Bitcoin space. TBH even the original comment wasn't really necessary.

Reply:
\me hangs head. I need to spend more time in the Bitcoin space.